plugins/Overlay/templates/index.twig

```diff
@@ -74,6 +74,8 @@
             var iframeSrc = 'index.php?module=Overlay&action=startOverlaySession&idSite={{ idSite }}&period={{ period }}&date={{ rawDate }}&segment={{ segment }}';
             if (piwik.shouldPropagateTokenAuth) {
                 iframeSrc += '&force_api_session=1&token_auth=' + piwik.token_auth;
+            } else if (piwik.token_auth.length) {
```

Comment:
I think here we would need to check if the original request includes the `&token_auth` parameter in the URL as part of `$_GET`. I haven't tested but I'm thinking otherwise we will always append the `token_auth` parameter into the URL even when the user is requesting it without a token. That's because `piwik.token_auth` is always set.

I reckon in the controller action `Overlay.index`  we could maybe check if the token_auth parameter is set in the GET request and if so, only then fall back to this logic.

Reply:
fyi also in the line above (in the `if (piwik.shouldPropagateTokenAuth) {`) we will need to add the check for `force_api_session` like:

```
 if (!piwik.broadcast.isWidgetizeRequestWithoutSession()) {
                url += '&force_api_session=1';
            }
```

Reply:
We could maybe also adjust `View::shouldPropagateTokenAuthInAjaxRequests()` to add the token auth when the token is present but not sure if this could have side effects

Reply:
All problems seem to be resolved now, @tsteur .